### ISSUE TITLE
RETURN key occasionally injects gibberish

### ISSUE THREAD

**Issue by brodders61** [spyder-ide/spyder#4166]
## Description of your problem

Under Windows 7 (seen also on 8.1) using any 3 series Spyder

1. Use editor
2. approximately 1 in 100 uses of RETURN key results in corruption e.g.
newline not inserted
some random text (including backspaces) inserted
3. Ctrl-Z still works

Example:
 typing just anything then hit RETURN for newline here

_usually works but sometimes gets=>>_

 typing just anything then hit RETUR): # old stuff comment from who knows where
sometime goes onto this line too

## Versions and main components

* Spyder Version: 3.1.2
* Python Version: 2.7.13
* Qt Version: 5.6.2
* PyQt Version: 5.6
* Operating system: Win 7

## Dependencies

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  2.0.0 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.3 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.5.0 :  1.5.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
sympy >=0.7.3    :  1.0 (OK)

**Comment by brodders61** [comment]
Example just happened (been seeing this for weeks with different Spyder versions...)

Cut and paste text to make a variant def(). Here is the copied original:
```
def make_feeder_kW_staticloads(house_phase_load_LoL):

    for period_idx in range(0, len(house_phase_load_LoL[0])):

        all_phase_loads_Wh = house_phase_load_LoL[1][period_idx] + \
                             house_phase_load_LoL[2][period_idx] + \
                             house_phase_load_LoL[3][period_idx]
```
On the copy, doing changes to top line so had:

```
def make_nominal_feeder_kW_forecast(per_st_delta, house_phase_load_LoL):

    for period_idx in range(0, len(house_phase_load_LoL[0])):

```
with insert cursor after : on the def line press RETURN key, and this comes out:

```
def make_nominal_feeder_kW_forecast(per_st_delta, house_phaseper_st_delta
    for period_idx in range(0, len(house_phase_load_LoL[0])):

```
SO: what's happened there? Editor...
did 11 backspaces
injected arbitrary text (??) "per_st_delta" and
did enough DEL's to remove the newline before the "for".

**Comment by brodders61** [comment]
Not used to this system and though 'Close' would close the session, not the issue...

**Comment by ccordoba12** [comment]
Sorry for the confusion. We had some problems with auto-indentation in 3.1.2 but they were fixed in 3.1.3.

Please update to that version (I can't reproduce your error in 3.1.3).
